redisearch/schema.go

```diff
@@ -35,6 +35,42 @@ type Options struct {
 	// If the list is nil the default stop-words list is used.
 	// See https://oss.redislabs.com/redisearch/Stopwords.html#default_stop-word_list
 	Stopwords []string
+
+	// If set to true, creates a lightweight temporary index which will expire after the specified period of inactivity.
+	// The internal idle timer is reset whenever the index is searched or added to.
+	// Because such indexes are lightweight, you can create thousands of such indexes without negative performance implications.
+	Temporary       bool
+	TemporaryPeriod int
+}
+
+func NewOptions() *Options {
+	var opts = DefaultOptions
+	return &opts
+}
+
+// If set to true, creates a lightweight temporary index which will expire after the specified period of inactivity.
+// The internal idle timer is reset whenever the index is searched or added to.
+// To enable the temporary index creation, use SetTemporaryPeriod(). This method should be preferably used for disabling the flag
+func (options *Options) SetTemporary(temporary bool) *Options {
+	options.Temporary = temporary
+	return options
```

Comment:
Not sure someone will create a temporary object and then regret ... but it's ok to have this api I guess..

Reply:
I believe it makes sense if you want to reuse options across several creations. WDYT?